pkg/R/serialize.R

```diff
@@ -16,10 +16,11 @@ writeObject <- function(con, object, writeType = TRUE) {
   # NOTE: In R vectors have same type as objects. So we don't support
   # passing in vectors as arrays and instead require arrays to be passed
   # as lists.
+  type <- class(object)[[1]]  # class of POSIXlt is c("POSIXlt", "POSIXt")
   if (writeType) {
-    writeType(con, class(object))
+    writeType(con, type)
   }
-  switch(class(object),
+  switch(type,
          integer = writeInt(con, object),
          character = writeString(con, object),
          logical = writeBoolean(con, object),
```

Comment:
So it turns out `class(NA)` returns `logical` in R. So in `writeBoolean` we can add a `is.na` check and write out the same NullType for SparkSQL